docs/rules/jsx-filename-extension.md

```diff
@@ -22,6 +22,18 @@ function MyComponent() {
 
 ## Rule Options
 
+### `allow` (default: `"always"`)
+
+When to allow a JSX filename extension. By default all files may have a JSX extension. Set this to `as-needed` to only allow JSX file extensions in files that contain JSX syntax.
```

Comment:
As commented in the linked issue, I'm not convinced this is a safe option to add. `.js` files are only supposed to be JS - jsx syntax isn't the only thing that would make a file not be standard JS. For example, if it used any pre-stage-3 features, like decorators or something.

At the very least, the docs here should add another paragraph explaining the reasons why this might be contentious.

Reply:
Done.